lib/upgrade_analyzer.rb

```diff
@@ -0,0 +1,11 @@
+$LOAD_PATH.unshift(File.dirname(__FILE__))
```

Comment:
Isn't `lib` normally in the load path?  I normally consider changing `$LOAD_PATH` to be a sign of something that shouldn't be done that way...